README.md
robotnik
========

The Robotnik Project


Level 1/changeLog.txt
************************************
12.09.2014 - VERSION 0.0002 - STEINI
************************************
->  Tür kann jetzt geöffnet werden
    -> keine Überprüfung durch hitTest, sondern durch Standard-Abfrage der Position
    
->  GAME_VERSION Variable wird nun unten rechts im Bildschirm angezeigt
    -> bei Änderungen am besten Aktualisieren (auch hier im changeLog.txt)
    
->  Fenster mit der Ladeanzeige für assets scrolled nun automatisch ans Ende, sobald ein
    neues Asset geladen wird
->  Schriftgrösse der Ladeanzeige angepasst

->  Utilitie-Funktionen nun ausgelagert nach ../lib/robotnik.js und aus den einzelnen logik.js entfernt 
    Sorgt für bessere Übersichtlichkeit (auch in Level 2)


************************************
28.08.2014 - VERSION 0.0001 - FELIX 
************************************

->  Einführung der Versionsnummern und Changelog

lib/robotnik.js
function debug() {
    document.getElementById("FPS").innerHTML = "Debug-Information:<br>"+Math.round(createjs.Ticker.getMeasuredFPS(),2)+"FPS";
    
}

function removeDebug() {
    document.getElementById("debug").style.visibility = "hidden";
    console.log("Debug-Information is hidden now!");
}

function showMessage(message,type) {
    document.getElementById("overlayMessage").style.visibility = "visible";
    document.getElementById("overlayMessage").style.fontFamily = "Ubuntu";
    document.getElementById("overlayMessage").style.width = screen.width;
    if(type == "error")
        document.getElementById("overlayMessage").style.backgroundColor = "#c84205";
    else if(type == "info")
        document.getElementById("overlayMessage").style.backgroundColor = "#079b00";
    else if(type == "load")
        document.getElementById("overlayMessage").style.backgroundColor = "#D8E7FF"; 
    else if(type == "changelog"){
        document.getElementById("overlayMessage").style.backgroundColor = "white";
        document.getElementById("overlayMessage").style.fontFamily = "Monospace";
    }
        
    document.getElementById("overlayMessage").innerHTML = message;
}

function scrollOlMessage() {
    theOverlay = document.getElementById("overlayMessage");
    console.log("Height of overlay: " + theOverlay.scrollHeight);
    scrPos = theOverlay.scrollHeight + 100;
    theOverlay.scrollTop = scrPos;
    console.log("New Pos: " + scrPos);
}

function hideMessage() {
    document.getElementById("overlayMessage").style.visibility = "hidden";
}

function changeMusicVolume() {
    volume = document.getElementById("music").value*0.01;
    
    MUSIC_VOLUME = volume;
    backgroundMusic.setVolume(volume);
}

function changeFXVolume() {
    volume = document.getElementById("fx").value*0.01;
    
    FX_VOLUME = volume;
    console.log(FX_VOLUME);
}

function setGameVersion() {
    document.getElementById("footer").innerHTML = "Release: <br>"+ GAME_VERSION + "<br><button>Show ChangeLog.txt</button>";
}

function showChangeLog() {
    showMessage(GAME_VERSION+'<br><iframe src="changeLog.txt" data="text/html" width="90%" height="260px" scrolling="yes" charset="UTF-8"></iframe>', "changelog");
    
    /*
    xmlhttp = new XMLHttpRequest();
    xmlhttp.onreadystatechange=function()
    {
        if (xmlhttp.readyState==4 && xmlhttp.status==200)
        {
            
            showMessage(xmlhttp.responseText, "changelog");
        }
    }
    xmlhttp.open("GET","changeLog.txt",true);
    xmlhttp.send();
    */
}

Level 2/logik001.js
var GAME_VERSION="ver 0.00001 - 28.8.14, 18:00";

var stage, background, waste, box, boxOpen, key, door, queue, loaded = "", backgroundMusic, ventil;
var wasteSND, lockedDoorSND, ventilSND, unlockSND;
// game logic 
var isWasteCleared=false;
var isBoxOpen=false;
var isDoorLocked=true;
var isDoorOpen=false;

// important for key drag and drop
var isDraggingKey=false;
    
const BOX_X=640,BOX_Y=205;
const X_CLEARED= 640;

const KEY_X=BOX_X+65, KEY_Y=BOX_Y+50;
const DOOR_X=815,DOOR_Y=525;
var FX_VOLUME = 0.6, MUSIC_VOLUME = 0.8;

function preloadAssets() {
    queue = new createjs.LoadQueue();
    
        
    // enables local development: deactivate for web Build - felix
    queue.setUseXHR(false);
    
    queue.installPlugin(createjs.Sound);
    queue.addEventListener("fileload", handleFileLoad);
    queue.addEventListener("fileprogress", updateProgress);
    queue.addEventListener("complete", handleComplete);
    queue.addEventListener("error", handleError);
    
    queue.loadManifest([
        {id:"background", src:"assets/background.png"},
        {id:"waste", src:"assets/waste.png"},
        {id:"boxOpen", src:"assets/boxOpen.png"},
        {id:"boxClosed", src:"assets/boxClosed.png"},
        {id:"key", src: "assets/schluessel.png"},
        {id:"door", src: "assets/door.png"},
        {id:"music", src:"assets/Music.mp3"},
        {id:"doorOpens", src:"assets/openDoor.wav"},
        {id:"doorCloses", src:"assets/shutDoor.mp3"},
        {id:"keySound", src: "assets/keySound.mp3"},
        {id:"ventil", src: "assets/ventil.png"},
                        
        {id:"lockedDoorSND", src: "assets/lockedDoor.wav"},
        {id:"wasteSND", src: "assets/moveDebris.wav"},
        {id:"ventilSND", src: "assets/ventil.wav"},
        {id:"unlockSND", src: "assets/unlock.wav"}

    ]);
}
function updateProgress(event) {
    document.getElementById("loadStatus").innerHTML = Math.round(event.progress*100,2) + "% loaded";
    //showMessage(Math.round(event.progress*100,2) + "% loaded", "load");
}
function handleFileLoad(event) {
    console.log("Loading " + event.item.id + " of type " + event.item.type);
    
    // beautiful switch-case for our loader - Felix
    
   switch(event.item.id){
       case "background":
           background = new createjs.Bitmap( event.result ); break;
       case "waste":
           waste = new createjs.Bitmap(event.result); break;
       case "boxOpen":
           boxOpen = new createjs.Bitmap(event.result);break;
       case "boxClosed":
           box = new createjs.Bitmap(event.result);break;
       case "key":
           key = new createjs.Bitmap(event.result);break;
       case"door":
           door = new createjs.Bitmap(event.result);break;
       case "ventil":
           ventil = new createjs.Bitmap(event.result);break;
           
    }
       
    loaded += "Asset [" + event.item.id + "] of type [" + event.item.type + "]"+"<br>";
    showMessage(loaded, "load");
}

function handleError(event) {
    console.error("There was an error loading " + event.item.id + " [type: "+event.item.type+"]");
}

function handleComplete() {
    document.getElementById("loadStatus").innerHTML = "All assets loaded successfully!";
    hideMessage();
    init();
}

function init() {
    initializeObjects();
    backgroundMusic = createjs.Sound.play("music",{loop:-1});
    backgroundMusic.setVolume(MUSIC_VOLUME);
    
    document.getElementById("music").value = MUSIC_VOLUME*100;
    document.getElementById("fx").value = FX_VOLUME*100;
    
    

    box.on("click", function(evt){
                
        if(isWasteCleared) {                   
            if(!isBoxOpen){                      
                openBox();              
            }               
        }              
        else {                 
            //showMessage("Oh no, the door is blocked by some waste. Clean up first!<br>-.-", "error"); 
            wobbleBox();
            

            
        } 
    
    });
    
    function wobbleBox() {
        createjs.Sound.play("wasteSND").setVolume(FX_VOLUME);
        
            newX = waste.x - 10;
            oldX = waste.x;
            console.log(waste.x + "-"+newX + "-" + oldX);

            createjs.Tween.get(waste)
                .to({x: newX},150)
                .wait(50)
                .to({x: oldX}, 150);
    
    
    boxOpen.on("click", function(evt){        
        if(isBoxOpen)closeBox();        
    })
    
    //
    // key
    
    
    key.on("pressmove",function(evt){
        evt.currentTarget.x=evt.stageX;
        evt.currentTarget.y=evt.stageY;
    })
    
    key.on("pressup",function(evt){        
        if(door.hitTest(evt.X,evt.Y)) unlockDoor();
        else returnKey();    
    })
    
    //
    // waste
    
    waste.on("mousedown",function(evt){
        createjs.Sound.play("wasteSND").setVolume(FX_VOLUME);
    })
    
    waste.on("pressmove",function(evt) {
 
        
            if(!isWasteCleared){                
                evt.currentTarget.x = evt.stageX;
                    if(evt.stageX < 600)        
                    evt.currentTarget.x = 600;
            
                if(evt.stageX > 730)
                evt.currentTarget.x = 730;
            //evt.currentTarget.y = evt.stageY;
   
    
            //stop dragging, when box is clear
                //Brauchen wir das wirklich? -steini
            if(evt.currentTarget.x<X_CLEARED){
                
                isWasteCleared=true;
            }
            }});
    
    waste.on("pressup", function() {
        if(!isWasteCleared) {
            wobbleBox();
        }
    });
    
    
    door.on("click",function(evt){
        if(isDoorLocked)        createjs.Sound.play("lockedDoorSND").setVolume(FX_VOLUME);
        else openDoor();}
        )
    
    ventil.on("click",function(evt){
        createjs.Sound.play("ventilSND").setVolume(FX_VOLUME);
    })
    
    
    
    
}

//
// initializers - graphic
//
function initializeObjects(){
    initializeStage();
    initializeBoxes();
    initializeWaste();
    initializeKey();
    initializeDoor();
    
    initializeVentil();
    
    stage.addChild(background,box,waste,door,ventil);
}


function initializeStage(){
    stage = new createjs.Stage(document.getElementById("canvas"));
    createjs.Ticker.addEventListener("tick", tick);
    createjs.Ticker.setFPS(60);
    createjs.Ticker.timingMode = createjs.Ticker.RAF_SYNCED;
}

function initializeBoxes(){
    //box = new createjs.Bitmap("assets/boxClosed.png");
    box.x = BOX_X;
    box.y = BOX_Y;
                   
    //boxOpen = new createjs.Bitmap("assets/boxOpen.png");              
    boxOpen.x=BOX_X;               
    boxOpen.y=BOX_Y;
}

function initializeWaste(){
    //waste = new createjs.Bitmap("assets/waste.png");
    waste.x = 720;
    waste.y = 287;
    waste.regX = 60;
    waste.regY = 42;
  
}

function initializeKey(){
    //key=new createjs.Bitmap("assets/schluessel.png");
    //Added regX, regY to center the key on dragging -steini
    key.regX = 8;
    key.regY = 25;
    key.x=KEY_X;
    key.y=KEY_Y;
}

function initializeDoor(){
    //door=new createjs.Bitmap("assets/door.png");
    door.x=DOOR_X;
    door.y=DOOR_Y;
    
}

function initializeVentil(){
    ventil.x=576;
    ventil.y=444,5;
}

//
// game functions
//

function openBox(){     
    createjs.Sound.play("doorOpens").setVolume(FX_VOLUME);
    stage.removeChild(box);    
    stage.addChild(boxOpen); 
    stage.addChild(key);
    isBoxOpen=true;
}

function closeBox(){    
    createjs.Sound.play("doorCloses").setVolume(FX_VOLUME);
    stage.removeChild(boxOpen);
    stage.removeChild(key);
    stage.addChild(box);
    isBoxOpen=false;
}
    
    function openDoor(){
        createjs.Sound.play("doorOpens").setVolume(FX_VOLUME);
    }

function unlockDoor(){
    createjs.Sound.play("unlockSND").setVolume(FX_VOLUME);
    isDoorLocked=false;
}

function returnKey(){
    createjs.Sound.play("keySound").setVolume(FX_VOLUME);
    createjs.Tween.get(key)
        .to({x:KEY_X, y:KEY_Y},500,createjs.Ease.backInOut)
        .call(closeBox); 
}
    
    
//
// utilities
//



//
//
//
    

function tick(event) {
    stage.update();
    debug();
}

function debug() {
    document.getElementById("FPS").innerHTML = "Debug-Information:<br>"+Math.round(createjs.Ticker.getMeasuredFPS(),2)+"FPS";
    
}

function removeDebug() {
    document.getElementById("debug").style.visibility = "hidden";
    console.log("Debug-Information is hidden now!");
}

function showMessage(message,type) {
    document.getElementById("overlayMessage").style.visibility = "visible";
    document.getElementById("overlayMessage").style.width = screen.width;
    if(type == "error")
        document.getElementById("overlayMessage").style.backgroundColor = "#c84205";
    else if(type == "info")
        document.getElementById("overlayMessage").style.backgroundColor = "#079b00";
    else if(type == "load")
        document.getElementById("overlayMessage").style.backgroundColor = "#D8E7FF";    
    console.log(screen.width);
    document.getElementById("overlayMessage").innerHTML = message;
}

function hideMessage() {
    document.getElementById("overlayMessage").style.visibility = "hidden";
}

function changeMusicVolume() {
    volume = document.getElementById("music").value*0.01;
    
    MUSIC_VOLUME = volume;
    backgroundMusic.setVolume(volume);
}

function changeFXVolume() {
    volume = document.getElementById("fx").value*0.01;
    
    FX_VOLUME = volume;
    console.log(FX_VOLUME);
}


Level 1/logik.js
var GAME_VERSION="ver 0.0001 - 4.9.14, 17:00, last edited by Felix";

var stage, background, loaded;

//sounds
var sheetSND, switchSND, electricSND,ventSND, backgroundMusic;
var FX_VOLUME = 0.6, MUSIC_VOLUME = 0.8;


//objects
var book, page1,page2,page3,page4,page5,notePaper,notePaperBig;

var boxClosed,boxOpen, boxGUI,box_X=305,box_Y=278,boxGUI_X=500,boxGUI_Y=200;
var swL1,swL2,swL3,swVentilator,swDoor, switch_X=200,switch_Y_init=200,switch_Height=200;


//shadows
var shLeft,shMid,shRight, darkMid,darkRight,darkLeft;

//gameVars
var isLightOn=false,hasOpenendBook=false,hasNotePaper=false, isUnlocked=false, isUnderVoltage=true, isBoxOpen=false;

//positions
const bookX=519,bookY=70,notePaperX=1000,notePaperY=0;




function preloadAssets() {
    queue = new createjs.LoadQueue();
        
    // enables local development: deactivate for web Build - felix
    queue.setUseXHR(false);
    
    queue.installPlugin(createjs.Sound);
    queue.addEventListener("fileload", handleFileLoad);
    queue.addEventListener("fileprogress", updateProgress);
    queue.addEventListener("complete", handleComplete);
    queue.addEventListener("error", handleError);
    
    queue.loadManifest([
        {id:"background", src:"assets/background.png"},
		{id:"shLeft", src:"assets/shadowLeft.png"},
		{id:"shMid", src:"assets/shadowMid.png"},
		{id:"shRight", src:"assets/shadowRight.png"},
		{id:"darkMid", src:"assets/darkMid.png"},
		{id:"darkRight", src:"assets/darkRight.png"},
		{id:"darkLeft", src:"assets/darkLeft.png"},
        {id:"book", src:"assets/book.png"},
        
                
        {id:"page1", src:"assets/page1.png"},
        {id:"page2", src:"assets/page2.png"},
        {id:"page3", src:"assets/page3.png"},
        {id:"page4", src:"assets/page4.png"},
        {id:"page5", src:"assets/page5.png"},
        {id:"notePaper", src:"assets/notePaper.png"},
        {id:"notePaperBig", src:"assets/notePaperBig.png"},
		
		{id:"boxClosed", src:"assets/boxShut.png"},
		{id:"boxOpen", src:"assets/boxOpen.png"},
		
		{id:"music", src:"assets/background.wav"},
		{id:"boxOpen", src:"assets/boxOpen.png"},
		{id:"boxOpen", src:"assets/boxOpen.png"},
		{id:"boxOpen", src:"assets/boxOpen.png"},
		{id:"boxOpen", src:"assets/boxOpen.png"},		
		
		


        
    ]);
}
function updateProgress(event) {
    document.getElementById("loadStatus").innerHTML = Math.round(event.progress*100,2) + "% loaded";
    //showMessage(Math.round(event.progress*100,2) + "% loaded", "load");
}
function handleFileLoad(event) {
    console.log("Loading " + event.item.id + " of type " + event.item.type);
    
    // beautiful switch-case for our loader - Felix
    
   switch(event.item.id){
       case "background":
           background = new createjs.Bitmap( event.result ); break;
		
		case "shLeft": 
			shLeft=new createjs.Bitmap( event.result ); break;
		
		case "shMid":
			shMid=new createjs.Bitmap( event.result ); break;
		
		case "shRight":
			shRight=new createjs.Bitmap( event.result ); break;
				
		case "darkMid":			
		darkMid=new createjs.Bitmap( event.result ); break;
			
		case "darkRight":		
		darkRight=new createjs.Bitmap( event.result ); break;
		case "darkLeft":		
		darkLeft=new createjs.Bitmap( event.result ); break;
		case "book":		
		book=new createjs.Bitmap( event.result ); break;
        
        case "page1":		
		page1=new createjs.Bitmap( event.result ); break;
		
        case "page2":		
		page2=new createjs.Bitmap( event.result ); break;
		
        case "page3":		
		page3=new createjs.Bitmap( event.result ); break;
		
        case "page4":		
		page4=new createjs.Bitmap( event.result ); break;
		
        case "page5":		
		page5=new createjs.Bitmap( event.result ); break;
		
        case "notePaper":		
		notePaper=new createjs.Bitmap( event.result ); break;
		
        case "notePaperBig":		
		notePaperBig=new createjs.Bitmap( event.result ); break;
		
		case "boxClosed":		
		boxClosed=new createjs.Bitmap( event.result ); break;
		
		case "boxOpen":		
		boxOpen=new createjs.Bitmap( event.result ); break;
		
           
    }
       
    loaded += "Asset [" + event.item.id + "] of type [" + event.item.type + "]"+"<br>";
    showMessage(loaded, "load");
}

function handleError(event) {
    console.error("There was an error loading " + event.item.id + " [type: "+event.item.type+"]");
}

function handleComplete() {
    document.getElementById("loadStatus").innerHTML = "All assets loaded successfully!";
    hideMessage();
    init();
}
		
		

		

function init() {
    initializeObjects();
	initializeSounds();
    setGameVersion();
		
    document.getElementById("music").value = MUSIC_VOLUME*100;
    document.getElementById("fx").value = FX_VOLUME*100;
		
	       
        
		
		book.on("click",function(evt){
            if (!hasOpenendBook){
            openBook();
            hasOpenendBook=true;
		
			sheetSND = createjs.Sound.play("assets/paper.wav");				
            }
        })
		
		// Bookpages
        
        page1.on("click",function(evt){
            stage.removeChild(page1);
		
			playSheetSound();
        })
        
                
        page2.on("click",function(evt){
            stage.removeChild(page2);
		
			playSheetSound();			
        })
        
                
        page3.on("click",function(evt){
            stage.removeChild(page3);
			
			playSheetSound();			
        })
        
                
        page4.on("click",function(evt){
            stage.removeChild(page4);
		
			playSheetSound();
			
        })
		
		//
        
        notePaper.on("click",function(evt){
            stage.removeChild(notePaper);

			playSheetSound();
		
            hasNotePaper=true;
            stage.addChild(notePaperBig);
            closeBook();
        })
		
				
		background.on("click",function(evt){
			switchLights();
			switchSND= createjs.Sound.play("assets/switch.wav").setVolume(FX_VOLUME);
		})
		
		
		
}

//
// initializers - graphic
//
function initializeObjects(){
    initializeStage();  
	initializeShadowLights();
    initializeDark();
	initializeGameObjects();
	initializeBox();
    
        
    //add Basic
	stage.addChild(background);
        stage.addChild(book,boxClosed);
	

	//add dark blends	
        stage.addChild(shLeft,shMid,shRight);
        stage.addChild(darkMid,darkRight);
}
		function initializeSounds(){
			backgroundMusic = createjs.Sound.play("music",{loop:-1});
		    backgroundMusic.setVolume(MUSIC_VOLUME);
			electricSND = createjs.Sound.play("assets/electric.wav",{loop:-1});
			electricSND.setVolume(FX_VOLUME);
			//ventSND = createjs.Sound.play("assets/ventilator.wav",{loop:-1});
		
			
		}



		function initializeStage(){    
		stage = new createjs.Stage(document.getElementById("canvas"));    
		createjs.Ticker.addEventListener("tick", tick);    
		createjs.Ticker.setFPS(60);    
		createjs.Ticker.timingMode = createjs.Ticker.RAF_SYNCED;
		}
		


		function initializeShadowLights(){
		shLeft.x=42;shLeft.y=82;
		shMid.x=353;shMid.y=94;
		shRight.x=657;shRight.y=126;
		}
		
		function initializeDark(){
		darkMid.x=shMid.x;
		darkMid.y=shMid.y;
		
		darkRight.x=shRight.x;
		darkRight.y=shRight.y;
		
		darkLeft.x=shLeft.x;
		darkLeft.y=shLeft.y;
		}
        
        function initializeGameObjects(){
        book.x=300;
        book.y=215;
        
        page1.x=bookX-4;page1.y=bookY+10;
        page2.x=bookX+4;page2.y=bookY-4;
        page3.x=bookX-10;page3.y=bookY+8;
        page4.x=bookX+2;page4.y=bookY-2;
        page5.x=bookX-11;page5.y=bookY+9;
        
        notePaper.x=bookX;notePaper.y=bookY;
        notePaperBig.x=notePaperX,notePaperBig.y=notePaperY;
        
        }
		
		function initializeBox(){

		boxClosed.x=box_X;boxClosed.y=box_Y;
		boxOpen.x=box_X,boxOpen.Y=box_Y;
		
		//boxClosed.rotation=120;
		}
    
//
// game functions
//
		
		function playSheetSound(){
			sheetSND.stop();			
			sheetSND.play();	
		}

        function openBook(){
        stage.addChild(page5,notePaper,page4,page3,page2,page1);
        stage.removeChild(book);
        }
        
        function closeBook(){
        stage.removeChild(page5);
        }
        
		function switchLights(){
		if(isLightOn){
		isLightOn=false;
		stage.addChild(darkMid,darkRight);
		}
		else{
		isLightOn=true;
		stage.removeChild(darkMid,darkRight);
		}
		}


//
//
//
    

function tick(event) {
    stage.update();
    debug();
}


		